lib/i18n.rb

```diff
@@ -371,7 +371,7 @@ def normalize_key(key, separator)
           keys.delete('')
           keys.map! do |k|
             case k
-            when /\A[-+]?[1-9]\d*\z/ # integer
+            when /\A(?:0|[-+]?[1-9]\d*)\z/ # integer
```

Comment:
I think `[-+]?[0-9]\d*` would also work here and would mean that we don't need to capture the expression. What do you think?

Reply:
Example on Rubular: https://rubular.com/r/N4d6XpMvk9GQhA

Reply:
It also would match 0001 and break the following test case for #457:
https://github.com/ruby-i18n/i18n/blob/e85bbb319ea30143e15986987fe1de5898273d4e/test/backend/simple_test.rb#L31-L35
So leading zeroes for multidigit integers should be forbidden.
Also `(?:…)` is a non-capturing group.